js/search-worker.js

```diff
@@ -42,11 +22,11 @@ function search(query) {
             var results = index.search(query);
 
             return results.map(function(result) {
-                return {
-                    score: result.score,
-                    url: result.ref,
-                    title: data[result.ref].title
-                };
+                // Extract the url and title from the ref string (they're separated by "|").
+                var i = result.ref.indexOf("|");
+                var url = result.ref.substring(0, i);
+                var title = result.ref.substring(i + 1);
```

Comment:
Is this going to cause a problem if there is a page somewhere without a title? Is it worth checking the length of `result.ref`?

Reply:
We do have pages that don't have titles and this code works for those cases. JavaScript's [`substring`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring) is pretty forgiving -- if you pass an index greater than or equal to the string length it returns an empty string, e.g. all the following cases return `""`:

```javascript
"/url|".substring(5);
"/url|".substring(6);
"/url|".substring(100);
```

I did add a check that throws if `"|"` isn't found.